src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/service/UserRoleService.kt

```diff
@@ -104,6 +104,11 @@ class UserRoleService(
     } ?: throw UsernameNotFoundException("User $userId not found")
   }
 
+  suspend fun getRolesByUsername(username: String): Set<Authority> {
+    userRepository.findByUsernameAndSource(username) ?: throw UsernameNotFoundException("User with username $username not found")
+    return roleRepository.findByUserRolesForUserName(username).toSet()
```

Comment:
Is this the query that you want, as this will default to auth users if you don't pass in a parameter